curifactory/staging.py

```diff
@@ -707,14 +846,92 @@ def wrapper(record: Record, records: list[Record] = None, **kwargs):
             utils.set_logging_prefix("")
             record.manager.stage_active = False
             record.manager.update_map_progress(record, "continue")
+            record.stage_cachers = None
             return record
 
         return wrapper
 
     return decorator
 
 
-def _check_cached_outputs(stage_name, record, outputs, cachers, records=None):
+def _get_output_representations_for_map(
+    record, outputs: list[str], cachers, records=None
+) -> list[MapArtifactRepresentation]:
+    """Check if the requested outputs are cached but do not load them, simply return
+    a list of statuses for found."""
+    if cachers == []:
+        raise EmptyCachersError(
+            "Do not use '[]' for cachers. This will always short-circuit because there is nothing that isn't cached."
+        )
+
+    artifacts = []
+    for i in range(len(outputs)):
+        metadata = None
+        cacher = None
+        is_cached = False
+        if cachers is not None:
+            cacher = cachers[i]
+            is_cached = cacher.check()
+            if is_cached:
+                metadata = cacher.load_metadata()
+        artifact = _add_output_artifact(
+            record, None, outputs, i, metadata, cacher, is_cached
+        )
+        artifacts.append(artifact)
+
+    return artifacts
+
+
+def _dag_skip_check_cached_outputs(
+    stage_name: str, record, outputs: list[str], cachers, records=None
+):
+    """Checks for cached values and loads any relevant metadata into artifact representations.
+    This function does not actually load values themselves, meant to be called for a skipped stage
+    in dag-based execution."""
+    if cachers == []:
+        raise EmptyCachersError(
+            "Do not use '[]' for cachers. This will always short-circuit because there is nothing that isn't cached."
+        )
+
+    skipped_function_outputs = []
+    for i, output_name in enumerate(outputs):
+        logging.debug("Checking cache status (not loading into memory.)")

```

Comment:
Do you want this debug logging statement to be output for every output? Should it go before the for loop?

Reply:
Not actually needed at all anymore, forgot to delete previously